src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -671,9 +671,28 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
             $dumper->setProxyDumper(new ProxyDumper(md5($cache->getPath())));
         }
 
-        $content = $dumper->dump(array('class' => $class, 'base_class' => $baseClass, 'file' => $cache->getPath(), 'debug' => $this->debug));
+        $content = $dumper->dump(array('class' => $class, 'base_class' => $baseClass, 'file' => $cache->getPath(), 'as_files' => !$this->debug, 'debug' => $this->debug));
```

Comment:
One of the strongest premises of the Symfony container was that it doesn't contain any significant differences between environments. This minimizes the WTF situations where the app works in dev but not in prod. Here it seems that we're breaking that important premise.